### PR TITLE
Add deployment script

### LINKED ISSUE #12: Add deployment script
### Description

Add `Dockerfile` and the github action setup to deploy on the staging environment

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #12 

### How was it solved?

* Add docker scripts
* Add deployment scripts

### How was it tested?

* build and run docker locally
* Run staging and production deployment scripts

### CODE REVIEW

File: scripts/generate_merkle_tree.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+if [ ! -d "$OUTPUT_DIR" ]; then
+    mkdir -p $OUTPUT_DIR
+else
+    echo "Output folder already exists"
+fi
+
+./packages/tree-builder/bin/run.js generate-merkle-tree --db-path=./tmp --output-path=$OUTPUT_DIR || exit 1
+
+# Upload files in $OUTPUT_DIR to S3 with folder token-claim
+aws s3 cp $OUTPUT_DIR s3://$S3_BUCKET_NAME/$FOLDER_NAME --recursive
```
Comment (matjazv) [inline scripts/generate_merkle_tree.sh]: Where is `S3_BUCKET_NAME` set?
Is this a private variable which behaves like a secret?

**Review by matjazv — CHANGES_REQUESTED** [review]
Why is this comment present?
`This branch had an error being deployed`

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,12 @@
+**/dist
+**/node_modules
+**/README.md
+**/LICENSE
+**/README.md
+**/tsconfig.eslint.json
+**/.eslintrc.json
+**/.mocharc.jsono
```
Comment (Phanco) [inline .dockerignore]: .mocharc.json?

File: scripts/generate_merkle_tree.sh (hunk 1 above)
Comment (shuse2) [reply scripts/generate_merkle_tree.sh]: S3_BUCKET_NAME is set as environment variable locally to keep it generic to put anywhere

**Comment by shuse2** [comment]
@matjazv it failed during test because the full setup was not ready

File: scripts/generate_merkle_tree.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+if [ ! -d "./tmp" ]; then
+    mkdir -p ./tmp
+fi
+
+if [ ! -f "./tmp/blockchain.db.tar.gz" ]; then
+    curl -o ./tmp/blockchain.db.tar.gz $SNAPSHOT_URL
+fi
+
+if [ ! -d "./tmp/blockchain.db" ]; then
```
Comment (matjazv) [inline scripts/generate_merkle_tree.sh:22]: Should be here `-f`?

**Review by matjazv — CHANGES_REQUESTED** [review]

File: scripts/generate_merkle_tree.sh (hunk 3 above)
Comment (shuse2) [reply scripts/generate_merkle_tree.sh:22]: blockchain.db is actually directory in the snapshot, and it contains blockchain.db and state.db 😢

**Review by matjazv — APPROVED** [review]

**Review by Phanco — APPROVED** [review]

**Pull request merged**